### ISSUE TITLE
cmd/go: TestScript fails on Windows, macOS with case-sensitive filesystem

### ISSUE THREAD

**Issue by lufia** [golang/go#30432]
My mac's disk is formatted with APFS that is case-sensitive variant.

### What version of Go are you using (`go version`)?

<pre>
$ go version
go version go1.12 darwin/amd64
</pre>

### Does this issue reproduce with the latest release?

Yes, bd98628 can reproduce this.

### What operating system and processor architecture are you using (`go env`)?

<details><summary><code>go env</code> Output</summary><br><pre>
$ go env
GOARCH="amd64"
GOBIN=""
GOCACHE="/Users/lufia/Library/Caches/go-build"
GOEXE=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="darwin"
GOOS="darwin"
GOPATH="/Users/lufia"
GOPROXY=""
GORACE=""
GOROOT="/Users/lufia/go"
GOTMPDIR=""
GOTOOLDIR="/Users/lufia/go/pkg/tool/darwin_amd64"
GCCGO="gccgo"
CC="clang"
CXX="clang++"
CGO_ENABLED="1"
GOMOD=""
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -fdebug-prefix-map=/var/folders/88/6vlybmvs1cv4zfjl4pcmjzdc0000gn/T/go-build789887819=/tmp/go-build -gno-record-gcc-switches -fno-common"
</pre></details>

### What did you do?

```console
$ git checkout go1.12
$ cd src
$ ./all.bash
```

### What did you expect to see?

I expect all tests are passed.

### What did you see instead?

```
$ ./all.bash 
Building Go cmd/dist using /Users/lufia/go1.9.
Building Go toolchain1 using /Users/lufia/go1.9.
Building Go bootstrap cmd/go (go_bootstrap) using Go toolchain1.
Building Go toolchain2 using go_bootstrap and Go toolchain1.
Building Go toolchain3 using go_bootstrap and Go toolchain2.
Building packages and commands for darwin/amd64.

##### Testing packages.
ok  	archive/tar	0.060s
ok  	archive/zip	2.737s
ok  	bufio	0.081s
ok  	bytes	0.427s
ok  	compress/bzip2	0.096s
ok  	compress/flate	1.145s
ok  	compress/gzip	0.030s
ok  	compress/lzw	0.019s
ok  	compress/zlib	0.030s
ok  	container/heap	0.026s
ok  	container/list	0.014s
ok  	container/ring	0.032s
ok  	context	1.031s
ok  	crypto	0.016s
ok  	crypto/aes	0.081s
ok  	crypto/cipher	0.582s
ok  	crypto/des	0.032s
ok  	crypto/dsa	0.025s
ok  	crypto/ecdsa	0.366s
ok  	crypto/elliptic	0.607s
ok  	crypto/hmac	0.018s
ok  	crypto/internal/subtle	0.012s
ok  	crypto/md5	0.013s
ok  	crypto/rand	0.046s
ok  	crypto/rc4	0.054s
ok  	crypto/rsa	0.100s
ok  	crypto/sha1	0.044s
ok  	crypto/sha256	0.012s
ok  	crypto/sha512	0.015s
ok  	crypto/subtle	0.012s
ok  	crypto/tls	1.737s
ok  	crypto/x509	2.954s
ok  	database/sql	0.611s
ok  	database/sql/driver	0.012s
ok  	debug/dwarf	0.030s
ok  	debug/elf	0.039s
ok  	debug/gosym	5.827s
ok  	debug/macho	0.015s
ok  	debug/pe	0.021s
ok  	debug/plan9obj	0.014s
ok  	encoding/ascii85	0.019s
ok  	encoding/asn1	0.031s
ok  	encoding/base32	0.031s
ok  	encoding/base64	0.020s
ok  	encoding/binary	0.016s
ok  	encoding/csv	0.017s
ok  	encoding/gob	0.066s
ok  	encoding/hex	0.016s
ok  	encoding/json	0.766s
ok  	encoding/pem	0.048s
ok  	encoding/xml	0.035s
ok  	errors	0.014s
ok  	expvar	0.039s
ok  	flag	0.016s
ok  	fmt	0.167s
ok  	go/ast	0.018s
ok  	go/build	0.435s
ok  	go/constant	0.013s
ok  	go/doc	0.092s
ok  	go/format	0.018s
ok  	go/importer	0.239s
ok  	go/internal/gccgoimporter	0.025s
ok  	go/internal/gcimporter	0.851s
ok  	go/internal/srcimporter	2.376s
ok  	go/parser	0.044s
ok  	go/printer	0.675s
ok  	go/scanner	0.014s
ok  	go/token	0.040s
ok  	go/types	1.476s
ok  	hash	0.014s
ok  	hash/adler32	0.020s
ok  	hash/crc32	0.016s
ok  	hash/crc64	0.021s
ok  	hash/fnv	0.013s
ok  	html	0.015s
ok  	html/template	0.074s
ok  	image	0.153s
ok  	image/color	0.038s
ok  	image/draw	0.092s
ok  	image/gif	0.595s
ok  	image/jpeg	0.463s
ok  	image/png	0.067s
ok  	index/suffixarray	0.017s
ok  	internal/cpu	0.043s
ok  	internal/fmtsort	0.010s
ok  	internal/poll	0.037s
ok  	internal/singleflight	0.023s
ok  	internal/trace	3.915s
ok  	internal/x/crypto/chacha20poly1305	0.145s
ok  	internal/x/crypto/cryptobyte	0.015s
ok  	internal/x/crypto/curve25519	0.040s
ok  	internal/x/crypto/hkdf	0.015s
ok  	internal/x/crypto/internal/chacha20	0.099s
ok  	internal/x/crypto/poly1305	0.019s
ok  	internal/x/net/dns/dnsmessage	0.105s
ok  	internal/x/net/http/httpguts	0.021s
ok  	internal/x/net/http/httpproxy	0.015s
ok  	internal/x/net/http2/hpack	0.020s
ok  	internal/x/net/idna	0.012s
ok  	internal/x/net/nettest	0.858s
ok  	internal/x/net/route	0.023s
ok  	internal/x/text/transform	0.015s
ok  	internal/x/text/unicode/norm	0.013s
ok  	internal/xcoff	0.044s
ok  	io	0.043s
ok  	io/ioutil	0.016s
ok  	log	0.019s
ok  	log/syslog	1.262s
ok  	math	0.018s
ok  	math/big	5.034s
ok  	math/bits	0.018s
ok  	math/cmplx	0.013s
ok  	math/rand	0.585s
ok  	mime	0.028s
ok  	mime/multipart	1.032s
ok  	mime/quotedprintable	0.168s
ok  	net	2.238s
ok  	net/http	4.464s
ok  	net/http/cgi	1.329s
ok  	net/http/cookiejar	0.056s
ok  	net/http/fcgi	0.038s
ok  	net/http/httptest	0.080s
ok  	net/http/httptrace	0.024s
ok  	net/http/httputil	0.115s
ok  	net/http/internal	0.010s
ok  	net/http/pprof	2.049s
ok  	net/internal/socktest	0.014s
ok  	net/mail	0.011s
ok  	net/rpc	0.049s
ok  	net/rpc/jsonrpc	0.034s
ok  	net/smtp	0.041s
ok  	net/textproto	0.013s
ok  	net/url	0.023s
ok  	os	1.110s
ok  	os/exec	2.167s
ok  	os/signal	4.781s
ok  	os/user	0.016s
ok  	path	0.015s
ok  	path/filepath	0.079s
ok  	plugin	0.010s
ok  	reflect	0.293s
ok  	regexp	0.107s
ok  	regexp/syntax	1.366s
ok  	runtime	34.544s
ok  	runtime/debug	0.096s
ok  	runtime/internal/atomic	0.212s
ok  	runtime/internal/math	0.031s
ok  	runtime/internal/sys	0.052s
ok  	runtime/pprof	12.546s
ok  	runtime/pprof/internal/profile	0.011s
ok  	runtime/trace	2.062s
ok  	sort	0.109s
ok  	strconv	1.571s
ok  	strings	0.430s
ok  	sync	0.376s
ok  	sync/atomic	0.042s
ok  	syscall	0.101s
ok  	testing	0.897s
ok  	testing/quick	0.224s
ok  	text/scanner	0.015s
ok  	text/tabwriter	0.015s
ok  	text/template	1.167s
ok  	text/template/parse	0.022s
ok  	time	2.694s
ok  	unicode	0.016s
ok  	unicode/utf16	0.018s
ok  	unicode/utf8	0.016s
ok  	cmd/addr2line	3.743s
ok  	cmd/api	0.029s
ok  	cmd/asm/internal/asm	1.083s
ok  	cmd/asm/internal/lex	0.014s
ok  	cmd/compile	0.017s
ok  	cmd/compile/internal/gc	26.152s
ok  	cmd/compile/internal/ssa	1.357s
ok  	cmd/compile/internal/syntax	0.025s
ok  	cmd/compile/internal/test	0.013s [no tests to run]
ok  	cmd/compile/internal/types	0.013s
ok  	cmd/cover	5.455s
ok  	cmd/doc	0.088s
ok  	cmd/fix	14.442s
go test proxy starting
go proxy_test: invalid module path encoding "example.com/invalidpath/v1": malformed module path "example.com/invalidpath/v1": invalid version
go test proxy running at GOPROXY=http://127.0.0.1:51600/mod
go proxy: no archive rsc.io v1.5.0
go proxy: no archive example.com/split/subpkg v1.0.0
go proxy: no archive golang.org/x/text/language 14c0d48
go proxy: no archive golang.org/x/text/language 14c0d48
go proxy: no archive golang.org/x/text/language 14c0d48
go proxy: no archive golang.org/x/text/foo 14c0d48
go proxy: no archive golang.org/x 14c0d48
go proxy: no archive golang.org 14c0d48
go proxy: no archive this.domain.is.invalid/somemodule v1.0.0
go proxy: no archive this.domain.is.invalid/somemodule v1.0.0
go proxy: no archive rsc.io/quote v1.99999999.0-not-a-real-version
--- FAIL: TestScript (0.00s)
    --- FAIL: TestScript/gcflags_patterns (3.45s)
        script_test.go:183: 
            # -gcflags=-e applies to named packages, not dependencies (0.091s)
            # -gcflags can specify package=flags, and can be repeated; last match wins (0.096s)
            # -gcflags can have arbitrary spaces around the flags (0.084s)
            # -gcflags='all=-e' should apply to all packages, even with go test (0.569s)
            # -ldflags for implicit test package applies to test binary (0.948s)
            # -ldflags for explicit test package applies to test binary (0.737s)
            # -ldflags applies to link of command (0.340s)
            # -ldflags applies to link of command even with strange directory name (0.299s)
            # -ldflags applies to current directory (0.270s)
            # -ldflags applies to current directory even if GOPATH is funny (0.000s)
            > [windows] cd $WORK/GoPath/src/my/cmd/prog
            > [darwin] cd $WORK/GoPath/src/my/cmd/prog
            FAIL: testdata/script/gcflags_patterns.txt:53: directory $WORK/GoPath/src/my/cmd/prog does not exist
            
FAIL
FAIL	cmd/go	163.021s
ok  	cmd/go/internal/cache	2.216s
ok  	cmd/go/internal/dirhash	0.020s
ok  	cmd/go/internal/generate	0.064s
ok  	cmd/go/internal/get	0.076s
ok  	cmd/go/internal/imports	0.019s
ok  	cmd/go/internal/load	0.051s
ok  	cmd/go/internal/lockedfile	0.031s
ok  	cmd/go/internal/lockedfile/internal/filelock	0.046s
ok  	cmd/go/internal/modconv	0.075s
ok  	cmd/go/internal/modfetch	0.187s
ok  	cmd/go/internal/modfetch/codehost	0.015s
ok  	cmd/go/internal/modfile	0.022s
ok  	cmd/go/internal/modload	0.091s
ok  	cmd/go/internal/module	0.013s
ok  	cmd/go/internal/mvs	0.018s
ok  	cmd/go/internal/par	0.044s
ok  	cmd/go/internal/search	0.017s
ok  	cmd/go/internal/semver	0.014s
ok  	cmd/go/internal/txtar	0.011s
ok  	cmd/go/internal/web2	0.041s
ok  	cmd/go/internal/work	0.033s
ok  	cmd/gofmt	0.186s
ok  	cmd/internal/buildid	0.422s
ok  	cmd/internal/dwarf	0.017s
ok  	cmd/internal/edit	0.014s
ok  	cmd/internal/goobj	1.623s
ok  	cmd/internal/obj	0.014s
ok  	cmd/internal/obj/arm64	0.086s
ok  	cmd/internal/obj/x86	0.819s
ok  	cmd/internal/objabi	0.013s
ok  	cmd/internal/src	0.016s
ok  	cmd/internal/test2json	0.159s
ok  	cmd/link	8.347s
ok  	cmd/link/internal/ld	4.513s
ok  	cmd/link/internal/sym	0.014s
ok  	cmd/nm	7.166s
ok  	cmd/objdump	5.990s
ok  	cmd/pack	4.383s
ok  	cmd/trace	0.070s
ok  	cmd/vendor/github.com/google/pprof/internal/binutils	0.048s
ok  	cmd/vendor/github.com/google/pprof/internal/driver	0.463s
ok  	cmd/vendor/github.com/google/pprof/internal/elfexec	0.015s
ok  	cmd/vendor/github.com/google/pprof/internal/graph	0.019s
ok  	cmd/vendor/github.com/google/pprof/internal/measurement	0.014s
ok  	cmd/vendor/github.com/google/pprof/internal/report	0.060s
ok  	cmd/vendor/github.com/google/pprof/internal/symbolizer	0.039s
ok  	cmd/vendor/github.com/google/pprof/internal/symbolz	0.042s
ok  	cmd/vendor/github.com/google/pprof/profile	0.099s
ok  	cmd/vendor/github.com/ianlancetaylor/demangle	0.034s
ok  	cmd/vendor/golang.org/x/arch/arm/armasm	0.032s
ok  	cmd/vendor/golang.org/x/arch/arm64/arm64asm	0.225s
ok  	cmd/vendor/golang.org/x/arch/ppc64/ppc64asm	0.015s
ok  	cmd/vendor/golang.org/x/arch/x86/x86asm	0.151s
ok  	cmd/vendor/golang.org/x/crypto/ssh/terminal	0.017s
ok  	cmd/vendor/golang.org/x/sys/unix	3.039s
ok  	cmd/vet	19.802s
2019/02/27 23:23:59 Failed: exit status 1
```

**Comment by bradfitz** [comment]
Also on Windows:
https://build.golang.org/log/0f0d0eba823c1fc90f40f67bf7427e909b8d2ccf

**Comment by bcmills** [comment]
@lufia, @bradfitz: the known-failing test has been fixed. Are there any other failures remaining for you?

**Comment by lufia** [comment]
@bcmills Thank you! I checked this issue is fixed in master branch (9e6ad46bccfa7a63e768236bcd1fd54dab38e4d1).

Note: **all.bash** still fails in `## GOOS=ios on darwin/amd64` section, but it is not related to this issue.

```
##### GOOS=ios on darwin/amd64
package crypto/x509
	imports crypto/x509/internal/macos: build constraints exclude all Go files in /Volumes/gotest/go/src/crypto/x509/internal/macos
2021/11/08 13:58:42 Failed: exit status 1
```

**Comment by bcmills** [comment]
Thanks for confirming!
